### PR TITLE
Update unambiguous converter to not support usernames

### LINKED ISSUE #2878: Update "unambiguous" converters given Discord username changes
The current logic for detecting a username in the unambigous username converter checks if it contains a discriminator, but since discord removed discriminators it doesn't make sense any more:

https://github.com/python-discord/bot/blob/da6997aab25b57152eec5d8c706977c472563853/bot/converters.py#L425-L433

We can probably safely just remove support for passing a username, since the converters are mostly used for moderation commands where IDs are pretty much all that is used anyway.

### PR DESCRIPTION
**Pull request by wookie184**
Closes #2878

### CODE REVIEW

**Comment by wookie184** [comment]
There are some accounts that still have discriminators so there are some other bits of code referring to discriminators that we might not want to change yet, but I don't think it really matters here.

**Review by vivekashok1221 — APPROVED** [review]
I haven't tested but code looks good to me.

**Comment by shtlrs** [comment]
I don't think it's quite used that much either but, just to be sure, how about we make a poll in mod-meta?

**Comment by wookie184** [comment]
> I don't think it's quite used that much either but, just to be sure, how about we make a poll in mod-meta?

I've asked there, though I assume if nobody has had an issue so far it'll be fine.

File: bot/converters.py [hunk 1]
```diff
@@ -423,23 +423,19 @@ async def convert(self, ctx: Context, argument: str) -> int:
 
 
 def _is_an_unambiguous_user_argument(argument: str) -> bool:
-    """Check if the provided argument is a user mention, user id, or username (name#discrim)."""
-    has_id_or_mention = bool(IDConverter()._get_id_match(argument) or RE_USER_MENTION.match(argument))
+    """Check if the provided argument is a user mention or user id."""
+    user_id = IDConverter()._get_id_match(argument)
```
Comment (shtlrs) [inline bot/converters.py:427]: `_get_id_match` is a static method, so no need to instantiate the converter.

**Review by shtlrs — APPROVED** [review]
Looks good & works well, thanks !

**Pull request merged**